### PR TITLE
[skip ci] Split part of the readme into CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by BlaineEXE**
The readme after all of the great documentation we wrote for the new
project structure was huge. Split it into a readme useful for most
people who want to casually use/understand the project and a new
`CONTRIBUTING.md` aimed at developers.

Signed-off-by: Blaine Gardner <blaine.gardner@suse.com>

### CODE REVIEW

**Comment by BlaineEXE** [comment]
I think @leseb might also want to verify that the content under the **CI** heading is still correct. Adding `[skip ci]` to the PR name is undocumented. Does it still work by commit? Is jenkins still in transition?

**Comment by leseb** [comment]
@BlaineEXE Jenkins is still in transition, the `[skip ci]` prefix works on the title of the PR only. If you have only one commit in your PR and if your commit has `[skip ci]` then this will work too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+   to avoid this as much as possible and focus on code reuse from the less specific dirs.
+4. Build test versions of the images (`make FLAVORS_TO_BUILD=<new distro flavors> build`).
+5. Test the images in your environment.
+6. Make a PR of your changes.
+
+
+CI
+-----
+
+We use Travis to run several tests on each pull request:
```
Comment (leseb) [inline CONTRIBUTING.md]: `s/Travis/Jenkins`, Travis is only used to test the `demo` scenario.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+For each PR, we try to detect which Ceph release is being impacted. Since we can only produce a
+single CI build with Travis, ideally this change will only be on a single release and distro. If we
+have multiple Ceph release and distro, we can only test one, since we have to build `base` and
+`daemon`. By default, we just pick up the first line that comes from the changes.
+
+You can check the files in `travis-builds` to learn more about the entire process.
+
+If you don't want to run a build for a particular commit, because all you are changing is the README
+for example, add `[skip ci]` to the git commit message. Commits that have `[skip ci]` anywhere in
+the commit messages are ignored by Travis CI.
```
Comment (leseb) [inline CONTRIBUTING.md]: `s/Travis/Jenkins`

PR with `[skip ci]` as part of their title.

**Pull request merged**